WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductRowCard.swift

```diff
@@ -116,6 +110,44 @@ struct CollapsibleProductRowCard: View {
     }
 }
 
+private extension CollapsibleProductRowCard {
+    @ViewBuilder var discountRow: some View {
+        if viewModel.discount == nil {
+            Button(Localization.addDiscountLabel) {
+                onAddDiscount()
+            }
+            .buttonStyle(PlusButtonStyle())
+        } else {
+            HStack {
+                Button(action: {
+                    onAddDiscount()
+                }, label: {
+                    HStack {
+                        Text(Localization.discountLabel)
+                        Image(uiImage: .pencilImage)
+                            .resizable()
+                            .frame(width: Layout.iconSize, height: Layout.iconSize)
+                    }
+                })
+                Spacer()
+                if let discountLabel = viewModel.discountLabel {
+                    Text(minusSign + discountLabel)
+                        .foregroundColor(.green)
+                }
+            }
+        }
+        Spacer()
+            .renderedIf(!viewModel.hasDiscount)
+        Button {
+            shouldShowCouponsInfoTooltip.toggle()
```

Comment:
For turning it _off_ again, I think we need more than just the button. I instinctively just tapped elsewhere on the screen and that just opens other modals

Reply:
Good catch, addressed in 61f28b59f35f213108d9489780dfbd72485ca27d and 61f28b59f35f213108d9489780dfbd72485ca27d . It's still not perfect, but covers most of the tappable screen.